core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -503,6 +536,10 @@ public void relocated(String reason) throws IllegalIndexShardStateException, Int
                         throw new IllegalIndexShardStateException(shardId, IndexShardState.STARTED,
                             ": shard is no longer relocating " + shardRouting);
                     }
+                    if (primaryReplicaResyncInProgress) {
```

Comment:
I missed that IllegalShardStateException is also handled as a delay in PeerRecoveryTargetService. All good.